test/should_fail/exhaustive_list_variants.erl

```diff
@@ -0,0 +1,12 @@
+-module(exhaustive_list_variants).
+
+-export([list_variants/1]).
+
+-type list_variant_t() :: {non_list, integer()}
+                        | {list, [integer()]}.
+
+-spec list_variants(list_variant_t()) -> ok.
+list_variants(T) ->
+    case T of
+        {non_list, _} -> ok
+    end.
```

Comment:
This seems to have nothing to do with lists. What am I missing?

Reply:
`list_variant_t()` has a variant which is basically a tagged list - hence its name. This tests exhaustiveness checking on types which have such variants. Well, the name could say `list_variants_exhaustiveness_test`, but it seemed self-explanatory - maybe it's not, if so I'll change the name.

Reply:
... but the list is just matched against the `_` pattern, which matches all types (lists or not) so I still can't see why any list-specific code would kick in...

Reply:
... apart from pick_value. Is that it?

Reply:
Yes, `typechecker:refinable()` and `gradualizer_lib:pick_value()`. Now I also extended this file to trigger exhaustiveness checking of (non)empty lists in the list variant.